#! /bin/sh -e

# gdc-arm-unwind_ptr.dpatch by Arthur Loiret <arthur.loiret@gmail.com>
# DP: Fix build on arm.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf2.59
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0


--- gcc/d/phobos/gcc/deh.d	2007-11-07 11:51:41.000000000 +0100
+++ gcc/d/phobos/gcc/deh.d	2007-11-07 11:51:54.000000000 +0100
@@ -254,7 +254,7 @@
     ubyte *p;
     ubyte *action_record;
     //int handler_switch_value;
-    _Unwind_Ptr /*landing_pad, */ip;
+    _Unwind_Ptr landing_pad, ip;
     Phase1Info phase1;
 
     static if (Use_ARM_EABI_Unwinder)
--- gcc/d/phobos2/gcc/deh.d	2007-11-07 18:05:26.000000000 +0100
+++ gcc/d/phobos2/gcc/deh.d	2007-11-07 18:05:55.000000000 +0100
@@ -254,7 +254,7 @@
     ubyte *p;
     ubyte *action_record;
     //int handler_switch_value;
-    _Unwind_Ptr /*landing_pad, */ip;
+    _Unwind_Ptr landing_pad, ip;
     Phase1Info phase1;
 
     static if (Use_ARM_EABI_Unwinder)
